libs/mutantschemata/source/mutantschemata/api.d

```diff
@@ -0,0 +1,151 @@
+/**
+Copyright: Copyright (c) 2019, Niklas Pettersson. All rights reserved.
+License: MPL-2
+Author: Niklas Pettersson (nikpe353@student.liu.se)
+
+This Source Code Form is subject to the terms of the Mozilla Public License,
+v.2.0. If a copy of the MPL was not distributed with this file, You can obtain
+one at http://mozilla.org/MPL/2.0/.
+
+/*
+discussed earlier, this api should handle:
+- filepath
+- mutant[]
+    - location
+    - mutant kind (operator)
+    - mutant_nr (injects)
+    - code of mutant
+
+Mutant schema provides:
+- Code
+- injects
+
+Meant to function as an api for schemata using and providing schemata with db
+*/
+module mutantschemata.api;
+
+import mutantschemata.d_string: cppToD, dToCpp;
+import mutantschemata.externals;
+import mutantschemata.utility: findInclude;
+
+import miniorm : Miniorm, buildSchema, delete_, insert, select;
+import dextool.type: Path;
+import dextool.plugin.mutate.backend.database: MutationPointTbl;
+import dextool.compilation_db: CompileCommandDB;
+
+import std.range: front;
+import std.array: array, empty;
+
+import logger = std.experimental.logger;
+
+// Entry point for Dextool mutate
+SchemataApi makeSchemataApi(Path db, CompileCommandDB ccdb) {
+    SchemataApi sa = new SchemataApi(db, ccdb);
+    return sa;
+}
+struct SchemataFileString {
+    string fpath;
+    SchemataMutant[] mutants;
+    string code;
+}
+SchemataFileString convert(SchemataFile sf) {
+    return SchemataFileString(cppToD!CppStr(sf.fpath),
+                                sf.mutants,
+                                cppToD!CppStr(sf.code));
+}
+
+// D class, connection to C++ code in /cpp_source
+class SchemataApi: SchemataApiCpp {
```

Comment:
I think it is enough if you add `extern(C++)` here. Should thus not need to put it on every method.